### PR TITLE
fix: stub every missing function in renderer package tests

### LINKED ISSUE #9288: Multiple errors like `TypeError: window.function-name is not a function` in `pnpm test:renderer`
### Bug description

There are multiple errors like one below when running `pnpm test:renderer` which indicates missing mocks for window funcitons.

```
Failed to update image checker providers TypeError: window.getImageCheckerProviders is not a function
    at EventStore.getImageCheckerProvidersInfo [as updater] (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/image-checker-providers.ts:42:17)
    at EventStore.performUpdate (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:142:35)
    at doUpdate (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:204:20)
    at update (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:209:15)
Failed to update views TypeError: window.listViewsContributions is not a function
    at EventStore.listViewsContributions [as updater] (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/views.ts:43:17)
    at EventStore.performUpdate (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:142:35)
    at doUpdate (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:204:20)
    at update (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:209:15)
Failed to update image files providers TypeError: window.getImageFilesProviders is not a function
    at EventStore.getImageFilesProvidersInfo [as updater] (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/image-files-providers.ts:42:17)
    at EventStore.performUpdate (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:142:35)
    at doUpdate (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:204:20)
    at update (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:209:15)
Failed to update config TypeError: window.getConfigurationProperties is not a function
    at EventStore.fetchConfigurationProperties [as updater] (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/configurationProperties.ts:47:85)
    at EventStore.performUpdate (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:142:35)
    at doUpdate (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:204:20)
    at update (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/stores/event-store.ts:209:15)
```

### Operating system

macOS Sequoia

### Installation Method

None

### Version

1.13.0

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dgolovin**
### What does this PR do?

Fix 'not a function' errors in renderer package tests

### What issues does this PR fix or reference?

Fix #9288.

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
AFAIK getConfigurationProperties is never returning undefined but `Record<string, IConfigurationPropertyRecordedSchema>`

so it should return `{}` not undefined

File: packages/renderer/src/lib/image/ImageDetails.spec.ts [hunk 1]
```diff
@@ -76,7 +76,10 @@ beforeEach(() => {
   (window as any).listContainers = vi.fn();
   (window as any).deleteImage = deleteImageMock;
   (window as any).hasAuthconfigForImage = hasAuthMock;
-
+  (window as any).getImageCheckerProviders = vi.fn().mockResolvedValue([]);
+  (window as any).listViewsContributions = vi.fn().mockResolvedValue([]);
+  (window as any).getImageFilesProviders = vi.fn().mockResolvedValue([]);
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue([]);
```
Comment (benoitf) [inline packages/renderer/src/lib/image/ImageDetails.spec.ts]: signature is not an array --> `{}`

File: packages/renderer/src/lib/image/ImagesList.spec.ts [hunk 2]
```diff
@@ -60,6 +60,8 @@ beforeEach(() => {
   (window as any).listViewsContributions = listViewsContributionsMock;
   listViewsContributionsMock.mockResolvedValue([]);
   (window as any).getConfigurationValue = vi.fn();
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue([]);
+  (window as any).deleteImage = vi.fn().mockResolvedValue(undefined);
```
Comment (benoitf) [inline packages/renderer/src/lib/image/ImagesList.spec.ts]: why do we need a resolved value ?

File: packages/renderer/src/lib/image/ImageDetails.spec.ts (hunk 1 above)
Comment (dgolovin) [inline packages/renderer/src/lib/image/ImageDetails.spec.ts]:
```suggestion
  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue({});
```

File: packages/renderer/src/lib/image/ImagesList.spec.ts [hunk 3]
```diff
@@ -60,6 +60,8 @@ beforeEach(() => {
   (window as any).listViewsContributions = listViewsContributionsMock;
   listViewsContributionsMock.mockResolvedValue([]);
   (window as any).getConfigurationValue = vi.fn();
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue([]);
```
Comment (dgolovin) [inline packages/renderer/src/lib/image/ImagesList.spec.ts]:
```suggestion
  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue({});
```

File: packages/renderer/src/lib/image/ImagesList.spec.ts (hunk 2 above)
Comment (dgolovin) [reply packages/renderer/src/lib/image/ImagesList.spec.ts]: 1. when getConfigurationProperties got mocked, deleteImage is not a function error appears on stdout
```
stderr | src/lib/image/ImagesList.spec.ts > Expect user confirmation to pop up when preferences require
error while removing image TypeError: window.deleteImage is not a function
    at ImageUtils.deleteImage (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/lib/image/image-utils.ts:245:19)
    at /Users/eskimo/Sources/podman-desktop/packages/renderer/src/lib/image/ImagesList.svelte:205:30
    at deleteSelectedImages (/Users/eskimo/Sources/podman-desktop/packages/renderer/src/lib/image/ImagesList.svelte:236:3)
```
2. when mocking deleteImage with vi.fn().mockResolvedValue() it requires one parameter

I can try if `vi.spyOn(window, 'deleteImage').mockResolvedValue();` works in tests, at least compiler does not complain about it.

Comment (benoitf) [reply packages/renderer/src/lib/image/ImagesList.spec.ts]: I mean, if we do only vi.fn(), we're checking the returning promise ?

File: packages/renderer/src/lib/container/ContainerDetails.spec.ts [hunk 4]
```diff
@@ -141,6 +141,8 @@ test('Expect show tty if container has tty enabled', async () => {
 });
 
 test('Expect redirect to previous page if container is deleted', async () => {
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue(undefined);
```
Comment (benoitf) [inline packages/renderer/src/lib/container/ContainerDetails.spec.ts]: issue: should not be undefined

File: packages/renderer/src/lib/compose/ComposeDetails.spec.ts [hunk 5]
```diff
@@ -54,7 +54,8 @@ beforeAll(() => {
       func();
     },
   };
-  (window as any).getConfigurationValue = vi.fn();
+  (window as any).getConfigurationValue = vi.fn().mockResolvedValue(undefined);
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue(undefined);
```
Comment (benoitf) [inline packages/renderer/src/lib/compose/ComposeDetails.spec.ts]: issue: no undefined, only {} is possible

File: packages/renderer/src/lib/pod/PodDetails.spec.ts [hunk 6]
```diff
@@ -70,6 +70,8 @@ beforeAll(() => {
   (window as any).getContributedMenus = getContributedMenusMock;
   (window as any).getConfigurationValue = getConfigurationValueMock;
   (window as any).addEventListener = vi.fn();
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue(undefined);
```
Comment (benoitf) [inline packages/renderer/src/lib/pod/PodDetails.spec.ts]: issue: can't be undefined

File: packages/renderer/src/lib/recommendation/ExtensionBanners.spec.ts [hunk 7]
```diff
@@ -36,6 +36,8 @@ beforeEach(() => {
   vi.resetAllMocks();
   (window as any).getConfigurationValue = getConfigurationValueMock;
   (window as any).getExtensionBanners = getExtensionBannersMock;
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValueOnce(undefined);
```
Comment (benoitf) [inline packages/renderer/src/lib/recommendation/ExtensionBanners.spec.ts]: issue: can't be undefined

File: packages/renderer/src/lib/volume/VolumeDetails.spec.ts [hunk 8]
```diff
@@ -58,6 +58,7 @@ beforeAll(() => {
   (window as any).showMessageBox = showMessageBoxMock;
   (window as any).listVolumes = listVolumesMock;
   (window as any).removeVolume = removeVolumeMock;
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue(undefined);
```
Comment (benoitf) [inline packages/renderer/src/lib/volume/VolumeDetails.spec.ts]: issue: can't be undefined

File: packages/renderer/src/lib/image/ImagesList.spec.ts (hunk 2 above)
Comment (dgolovin) [reply packages/renderer/src/lib/image/ImagesList.spec.ts]: that works too

File: packages/renderer/src/lib/compose/ComposeDetails.spec.ts (hunk 5 above)
Comment (dgolovin) [inline packages/renderer/src/lib/compose/ComposeDetails.spec.ts]:
```suggestion
  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue({});
```

Comment (dgolovin) [reply packages/renderer/src/lib/compose/ComposeDetails.spec.ts]: does not really matters for test, could be a sign of wrong exception handling, but it out of scope of the PR.

Comment (dgolovin) [inline packages/renderer/src/lib/compose/ComposeDetails.spec.ts:58]:
```suggestion
  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue({});
```

File: packages/renderer/src/lib/container/ContainerDetails.spec.ts (hunk 4 above)
Comment (dgolovin) [inline packages/renderer/src/lib/container/ContainerDetails.spec.ts]:
```suggestion
  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue({});
```

File: packages/renderer/src/lib/image/ImagesList.spec.ts [hunk 9]
```diff
@@ -60,6 +60,8 @@ beforeEach(() => {
   (window as any).listViewsContributions = listViewsContributionsMock;
   listViewsContributionsMock.mockResolvedValue([]);
   (window as any).getConfigurationValue = vi.fn();
+  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue({});
+  (window as any).deleteImage = vi.fn().mockResolvedValue(undefined);
```
Comment (dgolovin) [inline packages/renderer/src/lib/image/ImagesList.spec.ts]:
```suggestion
  (window as any).deleteImage = vi.fn();
```

File: packages/renderer/src/lib/pod/PodDetails.spec.ts (hunk 6 above)
Comment (dgolovin) [inline packages/renderer/src/lib/pod/PodDetails.spec.ts]:
```suggestion
  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue({});
```

File: packages/renderer/src/lib/recommendation/ExtensionBanners.spec.ts (hunk 7 above)
Comment (dgolovin) [inline packages/renderer/src/lib/recommendation/ExtensionBanners.spec.ts]:
```suggestion
  (window as any).getConfigurationProperties = vi.fn().mockResolvedValueOnce({});
```

File: packages/renderer/src/lib/volume/VolumeDetails.spec.ts (hunk 8 above)
Comment (dgolovin) [inline packages/renderer/src/lib/volume/VolumeDetails.spec.ts]:
```suggestion
  (window as any).getConfigurationProperties = vi.fn().mockResolvedValue({});
```

File: packages/renderer/src/lib/image/ImagesList.spec.ts (hunk 2 above)
Comment (dgolovin) [reply packages/renderer/src/lib/image/ImagesList.spec.ts]: updated

**Comment by dgolovin** [comment]
Updated all occurrences of using wrong mock for resolved value.

**Review by benoitf — APPROVED** [review]

**Pull request merged**